### ISSUE TITLE
Remove duplicate code from Navigation block

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#45033]
Refactor instances of duplicate code from the block where this has potential to cause regressions and bugs.

Let's not over optimise this but tackle the low hanging fruit only.

**Comment by amustaque97** [comment]
Hey @getdave, I would love to take a look at it. If no one is working on this I can take this up. 🙂

**Comment by getdave** [comment]
> Hey @getdave, I would love to take a look at it. If no one is working on this I can take this up. 🙂

@amustaque97 Absolutely. PRs are very welcome. The aim here is to remove obvious duplication. I don't think we need to take this too far and abstract everything because that may result in increased complexity, but there are opportunities for removing obviously repetitious code.

Thank you 🙇
